app/controllers/attachments_controller.rb

```diff
@@ -109,6 +103,15 @@ def authenticate_attachment
     redirect_to incoming_message_url(incoming_message), status: 303
   end
 
+  def authenticate_attachment_as_html
```

Comment:
I'm finding this a bit confusing.

It relies on @files_can_be_cached being set, and it's not really "authenticating" anyone. It requires `authenticate_attachment` to run before it, too, which is not obvious.

How about doing something more like:

```diff
   def cache_attachments
+    unless @files_can_be_cached
+      raise ActiveRecord::RecordNotFound, 'Attachment HTML not found.'
+    end
+
```

Reply:
Moving into the `cache_attachments` around action callback would require a bit more work. I think this can be clarified by renaming `s/files_can_be_cached/message_is_public/` and "authentication" the attachment makes more sense.